### ISSUE TITLE
Class 'OCA\DAV\Connector\Sabre\ExceptionLoggerPlugin' not found

### ISSUE THREAD

**Issue by awlx** [nextcloud/server#5043]
I upgraded from Nextcloud 11.3.0 to 12.0.0 stable via commandline. 
### Expected behaviour
Everything works like before

### Actual behaviour
Server is throwing 503 at me. 

### Server configuration
2017/05/22 17:49:21 [error] 13723#0: *545 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Class 'OCA\DAV\Connector\Sabre\ExceptionLoggerPlugin' not found in /var/www/user/web/nextcloud/remote.php on line 55" while reading response header from upstream, client: 192.168.0.135, server: cloud.user.org, request: "REPORT /nextcloud/remote.php/dav/calendars/user/shared/ HTTP/1.1", upstream: "fastcgi://127.0.0.1:9025"
**Operating system**:
Centos 6.9
**Web server:**
Nginx 1.10.2
**Database:**
5.5.54
**PHP version:**
5.6.30
**Nextcloud version:** (see Nextcloud admin page)
12.0.0
**Updated from an older Nextcloud/ownCloud or fresh install:**
11.3.0
**Where did you install Nextcloud from:**
Updater
**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - activity: 2.5.2
  - admin_audit: 1.2.0
  - bruteforcesettings: 1.0.2
  - comments: 1.2.0
  - dav: 1.3.0
  - encryption: 1.6.0
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - updatenotification: 1.2.0
  - workflowengine: 1.2.0
Disabled:
  - apporder
  - calendar
  - contacts
  - direct_menu
  - files_external
  - keeweb
  - notes
  - qownnotesapi
  - tasks
  - user_external
  - user_ldap
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>
{
    "system": {
        "appstore.experimental.enabled": true,
        "datadirectory": "\/opt\/nextcloud\/data",
        "dbhost": "127.0.0.1",
        "dbname": "c1owncloud",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbtype": "mysql",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "defaultapp": "apporder",
        "installed": true,
        "instanceid": "oc3fc1291743",
        "loglevel": 0,
        "mail_domain": "user.org",
        "mail_from_address": "cloud",
        "mail_smtpmode": "php",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\APC",
        "overwrite.cli.url": "\/owncloud",
        "overwritewebroot": "\/owncloud",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "trashbin_retention_obligation": "auto",
        "trusted_domains": [
            "cloud.user.org",
            "cloud.user.space"
        ],
        "updater.release.channel": "stable",
        "updaterRequirementsFulfilled": true,
        "version": "12.0.0.29"
    }
}
</details>

**Are you using encryption:** yes/no
yes

**Comment by awlx** [comment]
Found the issue for some reason Nextcloud was not able to connect to APC anymore changed the Variable to APCu and now it works.

**Comment by sbrl** [comment]
On updating from 17.0.1 to 17.0.2, I encountered this issue. I do _not_ have APC or APCu installed but had this auto-added to my config file:

```
memcache.local' => '\\OC\\Memcache\\APCu',
```

.....commenting it out fixed the issue.

You wouldn't expect a point release to cause such a serious issue (I got a nasty white screen with a plain text message), but apparently it did.

**Comment by kesselb** [comment]
Would you mind to report a new issue https://github.com/nextcloud/server/issues/new?labels=bug%2C+0.+Needs+triage&template=Bug_report.md?

I'm not aware of any code that changes `memcache.local` during updates. To triage this properly we need the issue template (including logs, installed apps, etc.). 

We usually don't reopen issues for such old versions. A new issue with up to date information is clearer. Thanks in advance :+1:

**Comment by sbrl** [comment]
Erm can do, @kesselb! I've fixed the issue now by following the step above though, so I'm not sure how much use it would be?

**Comment by juchong** [comment]
I also ran into this same issue. I'm happy to chime in on the eventual new issue if someone can point me to it.

**Comment by sbrl** [comment]
I could never reproduce the issue again @juchong, so I'd advise opening a new issue yourself since I'm not in a position to open a new issue anymore.

**Comment by JoshFialkoff** [comment]
I see this in the error log:
tack trace:
#0 /home/nextcloud/public_html/nextcloud/remote.php(169): handleException(Object(Doctrine\DBAL\DBALException))
#1 {main}
  thrown in /home/nextcloud/public_html/nextcloud/remote.php on line 61
[08-Jan-2021 21:10:02 UTC] PHP Fatal error:  Uncaught Error: Class 'OCA\DAV\Connector\Sabre\ExceptionLoggerPlugin' not found in /home/nextcloud/public_html/nextcloud/remote.php:61
Stack trace:
#0 /home/nextcloud/public_html/nextcloud/remote.php(169): handleException(Object(Doctrine\DBAL\DBALException))
#1 {main}
  thrown in /home/nextcloud/public_html/nextcloud/remote.php on line 61
[08-Jan-2021 21:20:10 UTC] PHP Fatal error:  Uncaught Error: Class 'OCA\DAV\Connector\Sabre\ExceptionLoggerPlugin' not found in /home/nextcloud/public_html/nextcloud/remote.php:61
Stack trace:
#0 /home/nextcloud/public_html/nextcloud/remote.php(169): handleException(Object(Doctrine\DBAL\DBALException))
#1 {main}
  thrown in /home/nextcloud/public_html/nextcloud/remote.php on line 61
[08-Jan-2021 21:25:07 UTC] PHP Fatal error:  Uncaught Error: Class 'OCA\DAV\Connector\Sabre\ExceptionLoggerPlugin' not found in /home/nextcloud/public_html/nextcloud/remote.php:61
Stack trace:
#0 /home/nextcloud/public_html/nextcloud/remote.php(169): handleException(Object(Doctrine\DBAL\DBALException))
#1 {main}
  thrown in /home/nextcloud/public_html/nextcloud/remote.php on line 61

What should I do?

**Comment by Sonnenbrand** [comment]
Same here

**Comment by juchong** [comment]
Hi folks,
I'm assuming that everyone here is using the Nextcloud Docker container? My issue went away after I pulled the latest image. Maybe something was pushed to main that was broken?

**Comment by Sonnenbrand** [comment]
> 
> 
> Hi folks,
> I'm assuming that everyone here is using the Nextcloud Docker container? My issue went away after I pulled the latest image. Maybe something was pushed to main that was broken?

Thanks for the advise, tried that but still no change!

**Comment by sirojinadias** [comment]
Same here, updating to PHP7.4 fixed it.

**Comment by satarsa** [comment]
I've got this error message after redis segfaulted, if it may help to track it.

**Comment by MrTomRod** [comment]
For me, the problem was that I somehow autoremoved `php7.4-apcu`.

Had to reinstall it: `apt-get install php7.4-apcu`

**Comment by Eschin** [comment]
This was totally related to redis not properly starting after 22.04 upgrade for me.

Fixing upgrade changes made in redis.conf and restarting fixed this for me.

**Comment by toneji** [comment]
> This was totally related to redis not properly starting after 22.04 upgrade for me.
> 
> Fixing upgrade changes made in redis.conf and restarting fixed this for me.

I also updated ubuntu 22.04 and got the same error, I am using it with Redis disabled.
If you don't mind, could you please tell me the contents of your Redis configuration file?

**Comment by Eschin** [comment]
> > This was totally related to redis not properly starting after 22.04 upgrade for me.
> > Fixing upgrade changes made in redis.conf and restarting fixed this for me.
> 
> I also updated ubuntu 22.04 and got the same error, I am using it with Redis disabled. If you don't mind, could you please tell me the contents of your Redis configuration file?

To clarify, my redis was starting, but attaching to a tcp port instead of a local unix socket :)

I'm running the stock 22.04 redis config, with these two changes;

- `port 0`
- `unixsocket /var/run/redis/redis-server.sock`

They were overwritten during the upgrade to 22.04 when I chose 'Y' to replace current config with package maintainer's config version.

I rebooted entire system after change, and nextcloud was back up and normal afterwards. But just restarting apache2 and redis services should be sufficient to bring it back up after correcting those redis settings.
